netbench/netbench/src/collector.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{scenario::Scenario, units::parse_duration, Result};
+use enum_dispatch::enum_dispatch;
+use std::time::Duration;
+use structopt::StructOpt;
+
+mod bpftrace;
+mod generic;
+mod procinfo;
+
+use bpftrace::BpftraceHandle;
+use generic::GenericHandle;
+
+#[derive(Debug, StructOpt, Clone)]
+pub struct Args {
+    pub driver: String,
+
+    #[structopt(long, short, env = "SCENARIO")]
+    pub scenario: String,
+
+    #[structopt(long, short, parse(try_from_str=parse_duration), default_value = "1s")]
+    pub interval: Duration,
+}
+
+impl Args {
+    pub fn scenario(&self) -> Result<Scenario> {
+        Scenario::open(std::path::Path::new(&self.scenario))
+    }
+}
+
+/// Use either the Bpftrace Handle or the Generic Handle
+#[enum_dispatch]
```

Comment:
What is enum dispatch, and why do we need it? This is feeling very java-lomboky, which I am not a fan of :D

Reply:
Hmm, doesn't seem that there is any particularly neat solution to this. Honestly I wonder if the solution is to just delete the Generic portion of the collector? I don't think it collects any particularly useful statistics, so that might be the route to go. Probably not something to address in this PR though.